### DISCUSSION TITLE
How do I install this?

### DISCUSSION THREAD

**Discussion by garydale** [SecKatie/ha-wyzeapi#433] [Q&A] [answered]
The documentation page for installation says "Add https://github.com/JoshuaMulliken/ha-wyzeapi as a custom repository as Type: Integration". In the home assistant add-on page I found a place to add a repository but when I try, it tells me that the URL is not a valid repository. I'm probably doing something wrong but without a more explicit instruction, I don't have a clue about what I should be doing.

**Comment by garydale** [comment]
After much pulling of hair and gnashing of teeth, I figured out how to do it. The instruction should say "On the Home Assistant sidebar select HACS then Integrations. From there, click on the "+ Explore & Download Repositories" button at the lower right. This will allow you to enter the URL for the wyzeapi repository. When asked, assign it as Type: Integration.

I find myself under whelmed by the integration in that it apparently is just a home assistant interface to the Wyze web site - a replacement for the smartphone app - rather than a tool to let you control the Wyze devices directly through Home Assistant. The difference being that it apparently:

1. logs into your Wyze account to adjust the settings for your devices rather than simply talking to the devices
2. this means the devices are actually controlled by Wyze rather than by Home Assistant.
3. this means you need Internet access and you expose your settings to Wyze.

I got into Home Assistant because I wanted a local solution to home automation since it's become difficult to find non-connected devices that do what I want (e.g. I had been controlling a light to turn on at dusk and off at a fixed time using a timer that estimated local sunrise and sunset based on your approximate latitude. It stopped working and I can't find a direct replacement). 

I tried a smartplug believing that any reasonable controller would keep things local. Wyze apparently doesn't do that and neither does the ha-wyzeapi.

**Comment by JoeSchubert** [comment]
Why would you think that an integration for a device set like this would somehow introduce local control to devices that doesn't exist in their firmware? I believe the bulbs (I own none) have had local control introduced, but still rely on being listed via the cloud'd API endpoint.  However, that's the only device in Wyze's lineup that supports local control as of yet. However, that should, in theory, change with matter support whenever they roll it out.

In short, there's not really anyway for any home assistant integration to force cloud only device to magically start allowing local control.

**Comment by JoeSchubert** [comment, marked as answer]
> "Add https://github.com/JoshuaMulliken/ha-wyzeapi as a custom repository as Type: Integration". In the home assistant add-on page I found a place to add a repository but when I try, it tells me that the URL is not a valid repository.

This isn't a home assistant add-on. It's a HACs addon repository. Once you install it through HACs, you restart HA and install it as a normal integration.

**Comment by spamoni4** [comment]
When I go to do the install via HACS, I receive the following "Repository 'seckatie/ha-wyzeapi' exists in the store."

My question is, What store?  I am using Home Assistant and for some reason not sure what is going on.  :)

**Comment by aero77** [comment]
ok so can I watch my cam thru the HA or get notifications from motion ?
